routes.jsx

```diff
@@ -15,21 +16,38 @@ import Navbar from './components/navbar/navbar.jsx';
 import Footer from './components/footer/footer.jsx';
 
 const App = React.createClass({
+  getInitialState() {
+    return {
+      suppressSplashScreen: localstorage.getItem(`suppressSplashScreen`) === `true`
+    };
+  },
   componentDidMount() {
-    setTimeout(function() {
-      this.splash.classList.add(`dismissed`);
-      document.querySelector(`#app`).classList.add(`splash-dismissed`);
-    }, 3000);
+    if (this.state.suppressSplashScreen !== `true`) {
+      setTimeout(function() {
+        this.splash.classList.add(`dismissed`);
+        document.querySelector(`#app`).classList.add(`splash-dismissed`);
+      }, 3000);
+      localstorage.setItem(`suppressSplashScreen`, `true`);
+    }
+  },
+  renderWelcomeSplash() {
+    if (this.state.suppressSplashScreen) {
+      return null;
+    }
+    return (
+      <div id="splash" ref={(splash) => { this.splash = splash; }}>
```

Comment:
that is a really weird thing to recommend. I can't say I agree with that at all: that DOM node has no business becoming permanently aliased at the component level, especially as it won't get cleaned up: when the splash render function returns `null`, what is `this.splash`? As such, I'd strongly recommend staying with `this.refs` instead:
```
if (this.refs.splash) {
  // assign class so it fades away
}
```
because there might not even be a splash screen to work with =)